src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+            }
+
+            if (conn.getGroupId() == null || conn.getGroupId() == 0) {
+                //Make sure UI sets group Id using getAgentDetails call.
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_GROUP_ID.getErrorMessage());
+            }
+
+            if (agentHostName == null || agentHostName.isEmpty()) {
+                //Make sure UI sets agentHostName using getAgentDetails call.
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_AGENT_HOSTNAME.getErrorMessage());
+            }
+
+            if (agentName == null || agentName.isEmpty()) {
+                //Make sure UI sets agentName using getAgentDetails call.
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_VALIDATING_AGENT_NAME.getErrorMessage());
+            }
+        }
+        try {
+            if (conn.getConnectionType() != null) {
+                conn.setModule(restDriver.getModuleTypeUsingSourceType(conn.getConnectionType()));
+                connectionId = restDriver.createConnection(conn);
+            } else {
+                throw new BadRequestException(
+                    ErrorConstants.ERROR_CREATING_CONNECTION.getErrorMessage());
+            }
+        } catch (DgDrvException e) {
+            logger.error("Create connection failed: " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Connection created with id: " + connectionId);
+
+        // Default schedule only for detection. In case of masking it is set to false from UI.
+        try {
+            if (defaultSchedule != null && defaultSchedule == true) {
+                SchedulerDTO schedule = getScheduleInfo(connectionId);
+                logger.info("Creating a schedule for connection with id: " + connectionId);
+                Boolean scheduleStatus = createDefaultDetectionSchedule(schedule, conn.getModule());
+                if (scheduleStatus == true) {
+                    logger.info("Default Schedule task for connection id " + connectionId
+                        + " completed successfully.");
+                } else {
+                    logger.error(
+                        "Default schedule task for connection id " + connectionId + " failed.");
+                }
+            } else {
+                logger.info("Default Schedule is null");
+            }
+        } catch (Exception e) {
+            logger.error("Error in creating schedule: " + e.getMessage());
+            logger
+                .info("Deleting the created connection quietly due to error in schedule creation");
+            try {
+                //deleteConnection(connectionId);
+                logger.info("Successfully deleted the connection.");
+            } catch (Exception e1) {
+                logger.error("Connection deletion failed: " + e1.getMessage());
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L186
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-148896979>*